### DISCUSSION TITLE
newb dev: compiling NI CVI

### DISCUSSION THREAD

**Discussion by ElectroLund** [microsoft/vscode-discussions#292] [Extension Development QnA] [answered]
Hey all, sorry for being such an extreme newb.  I've been using VS Code for C and AHK dev for some time now.  Loving it more and more!

I also develop code for a National Instruments platform called "LabWindows CVI."  It's an ANSI-C platform that adds extensions and libraries to build GUI-based apps that excel at talking to instruments and hardware (RS232, USB, serial, Ethernet, etc.).

I'd prefer to do my debugging with VS Code, rather than their IDE.  Where do I even start?  I'm assuming I would need to make my own VSC extension?  Is there a good howto for newbs?

**Comment by worksofliam** [comment, marked as answer]
Hey!

I can't speak for the IDE you're using and how that debugger works, but this is a nifty guide into how debugging works in the backend of VS Code:

https://code.visualstudio.com/api/extension-guides/debugger-extension
